courses/models.py

```diff
@@ -312,3 +312,35 @@ def get_freezable(cls):
         ).filter(freeze_grade_date__lt=now_in_utc())
 
         return course_runs
+
+
+class ElectivesSet(models.Model):
+    """
+    This represents an electives requirement for a program, with choice of courses and
+    required number of courses to be passed.
+    """
+    program = models.ForeignKey(Program, on_delete=models.CASCADE)
+    required_number = models.PositiveSmallIntegerField()
+    title = models.CharField(max_length=255)
+
+    def __str__(self):
+        return 'An electives set "{title}" for program "{program_title}"'.format(
+            title=self.title,
+            program_title=self.program.title
+        )
+
+
+class ElectiveCourse(models.Model):
+    """
+    Links to a course to an ElectivesSet
+    """
+    course = models.OneToOneField(Course, on_delete=models.CASCADE)
+    electives_set = models.ForeignKey(ElectivesSet, on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ('course', 'electives_set',)
+
+    def __str__(self):
+        return 'Course elective for course "{course_title}"'.format(
```

Comment:
I think this is a little clearer: 

```suggestion
        return 'Elective course "{course_title}" in electives set "{electives_set}"'.format(
```